packages/client/modules/outcomeCard/components/OutcomeCardAssignMenu/TaskFooterTeamAssigneeMenu.tsx

```diff
@@ -32,17 +60,57 @@ const TaskFooterTeamAssigneeMenu = (props: Props) => {
       : teams
     return filteredTeams
   }, [teamIds, userIds])
-  const taskTeamIdx = useMemo(() => assignableTeams.map(({id}) => id).indexOf(teamId) + 1, [
+  const taskTeamIdx = useMemo(() => assignableTeams.findIndex(({id}) => id === teamId) + 1, [
     teamId,
     assignableTeams
   ])
 
   const atmosphere = useAtmosphere()
   const {submitting, submitMutation, onError, onCompleted} = useMutationProps()
-  const handleTaskUpdate = (newTeam) => () => {
+
+  const onDialogClose = useEventCallback(() => {
+    menuProps.closePortal()
+  })
+  const {modalPortal, openPortal, closePortal} = useModal({
+    onClose: onDialogClose,
+    id: 'taskFooterTeamAssigneeAddGitHub',
+    parentId: 'taskFooterTeamAssigneeMenu'
+  })
+  const [newTeam, setNewTeam] = useState({id: '', name: ''})
+
+  const handleIntegrationAdded = () => {
+    closePortal()
+    if (!newTeam.id) return
+
+    submitMutation()
+    ChangeTaskTeamMutation(atmosphere, {taskId, teamId: newTeam.id}, {onError, onCompleted})
+    setNewTeam({id: '', name: ''})
+    menuProps.closePortal()
+  }
+  const handleClose = () => {
+    closePortal()
+    menuProps.closePortal()
+  }
+
+  const handleTaskUpdate = (newTeam) => async () => {
     if (!submitting && teamId !== newTeam.id) {
+      setNewTeam(newTeam)
+      if (isGitHubTask || isJiraTask) {
+        const result = await atmosphere.fetchQuery<
```

Comment:
I spend quite some time trying to make it work with `loadQuery`, but I wasn't successful. `loadQuery` is really only meant to be used in conjunction with `usePreloadedQuery`. It does not return any data if it doesn't do a network request.
From what I understand from [relay query-retention]](https://relay.dev/docs/guided-tour/reusing-cached-data/presence-of-data/#query-retention) the hooks would only retain data until they are going out of scope which would cause `dispose` to be called on the query. `dispose` should also be called on `loadQuery` thus the data retention would be rather limited.
I can wrap `fetchQuery` in a hook to also properly do the `environment.retain` and `dispose` calls, but using any of the other relay hooks requires much more state in the component* which I always find more difficult to read and debug.

*I would need to use the query in a hook (`useRefetchableFragment`) and use an effect on the `newTeam` state and the result of the hook, make sure it matches and then open the dialog. Not too bad, but I would like to avoid it. Also the queryReference would need to come from somewhere.

Reply:
you're not kidding! yikes, relay makes it _really_ hard to do that outside of react.

Interestingly enough, it looks like `fetchQuery` is doing what we want it to do. it keeps the query in the store & it _could_ be disposed of, but relay doesn't because our `gcReleaseBufferSize` is so large. Looks like you're right, fetchQuery is probably the best way to do things like this moving forward.